client/app/scripts/superdesk/editor/editor.js

```diff
@@ -725,11 +1107,74 @@ angular.module('superdesk.editor', ['superdesk.editor.spellcheck'])
 
                     scope.cursor = {};
                     render(null, null, true);
+
+                    function extractBlockContentsFromCaret() {
+                        function getBlockContainer(node) {
+                            while (node) {
+                                if (node.nodeType === 1 && /^(P|H[1-6]|DIV)$/i.test(node.nodeName)) {
```

Comment:
Maybe consider adding a comment that we test for paragraphs, divs and headings, makes it somewhat easier to read. :)